src/track/beatutils.cpp

```diff
@@ -12,37 +12,314 @@
+            continue;
+        } else {
+            // This may not be case when our median window is even, we can't use it
+            // Because find() will return an iterator pointing to end that we will *
+            if (tempoFrequency.contains(newStableTempo)) {
+                lastBeatChange = currentBeat - tempoMedianFilter.lag() - mostFrequentTempo.lag();
+                stableTemposByPosition[lastBeatChange] = newStableTempo;
+            }
+        }
+    }
+    stableTemposByPosition[tempoList.count()] = medianBpm;
+    return stableTemposByPosition;
+}
+
+void BeatUtils::RemoveSmallArrhythmic(
+        QVector<double>& rawBeats,  int sampleRate, QMap<int, double> &stableTemposByPosition) {
+    // A common problem the analyzer has is to detect arrhythmic regions
+    // of tacks with a constant tempo as in a different unsteady tempo. 
+    // This happens frequently on builds and breaks with heavy effects on edm music.
+    // Since these occurs most on beatless regions we do not want them to be
+    // on a different tempo, because they are still syncable in the true tempo
+    // We arbitraly remove these arrhythmic regions if they are short than 16s
+    auto positionsWithTempoChange = stableTemposByPosition.keys();
+    auto tempoValues = stableTemposByPosition.values();
+    QVector<double> anchoredBeats;
+    anchoredBeats.reserve(rawBeats.size());
+    anchoredBeats << QVector<double>::fromStdVector(std::vector<double>(
+                rawBeats.begin(), rawBeats.begin() + positionsWithTempoChange[1]));
+
+    for (int i = 2; i < positionsWithTempoChange.size(); i+=1) {
+        int smallInBeats =  16 / (60 / tempoValues[i-1]);
+        int limitAtLeft = positionsWithTempoChange[i-1];
+        int limitAtRight = positionsWithTempoChange[i];
+        int lenghtOfChange = limitAtRight - limitAtLeft;
+        double previousTempo = (stableTemposByPosition.lowerBound(limitAtLeft)-1).value();
+
+        if (lenghtOfChange <= smallInBeats) {
+            stableTemposByPosition.remove(limitAtLeft);
+            double beatOffset = rawBeats[limitAtLeft];
+            double beatLength = floor(((60.0 * sampleRate) / previousTempo) + 0.5);
+            while (beatOffset < rawBeats[limitAtRight]) {
+                anchoredBeats << beatOffset;
+                beatOffset += beatLength;
+            }
+
+        } else {
+            anchoredBeats << QVector<double>::fromStdVector(std::vector<double>(
+                    rawBeats.begin() + positionsWithTempoChange[i-1],
+                    rawBeats.begin() + positionsWithTempoChange[i]));
+        }
+    }
+    // We may have shinkred or enlarged our beat vector so we make sure our  
+    // last change sentinal does happens in the right place.
+    stableTemposByPosition.remove(stableTemposByPosition.last());
+    stableTemposByPosition[anchoredBeats.size() -1] = stableTemposByPosition[0];
+    rawBeats = anchoredBeats;
+}
+
+QVector<double> BeatUtils::FixBeatmap(
```

Comment:
Can we find a term? At least it should start with a lower case letter. 
I have no idea what he prefix "Fix" means

Reply:
smoothBeatmap?
ironBeatmap?
adjustBeatmap?

Reply:
Yes, I would pick "iron", because "smooth" sounds like just applying a smoothing filter. The opposite is true, like ironing a shirt. The fabrics smoothed, but the button bar and arms and so on are put in shape.

Reply:
changed